core/query/sql.ml

```diff
@@ -13,15 +13,21 @@ type query =
   | Insert    of {
       ins_table: table_name;
       ins_fields: string list;
-      ins_records: base list list
+      ins_records: insert_records
     }
   | Update    of {
       upd_table: table_name;
       upd_fields: (string * base) list;
       upd_where: base option
     }
   | Delete    of { del_table: table_name; del_where: base option }
-  | With      of table_name * query * query
+  | With      of table_name * query * query list
+  | Transaction of query list (* SQL Transaction: Complete atomically*)
```

Comment:
Yes, at present there's no support for transactions other than in the SQL DSL.

It would definitely be good to have some support for this. We'd probably need a better idea of exceptions, for a start, and I think after that we can do a design for purely update-based transactions. I wouldn't know the best design for then intertwining with regular and shredded queries, especially when relying things like `insert returning`, but I think it could be fun to look at.

Reply:
Tthe SQL queries resulting from a single shredded query ought to be wrapped in a transaction anyway, and aren't; we could do this now, using the new transaction case in Sql.query.  Having a design for source-level transactions seems like a separate (future) issue.